wagtail/admin/forms/account.py

```diff
@@ -115,7 +122,55 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self._original_avatar = self.instance.avatar
 
+    @contextmanager
+    def create_avatar_context(self, avatar):
+        avatar_context = willow.Image.open(avatar)
+        yield avatar_context
+
+    def reduce_avatar(self, avatar):
+        default_avatar_size = (400, 400)
+
+        with self.create_avatar_context(avatar) as willow_image:
+            width, height = willow_image.get_size()
+
+            should_reduce_avatar = (
+                width > default_avatar_size[0] or height > default_avatar_size[1]
+            )
+
+            if should_reduce_avatar:
+                temp_buffer = BytesIO()
+                original_format = (
+                    willow_image.format_name or willow_image.name.split(".")[-1].lower()
+                )
+                quality = getattr(settings, "WAGTAILIMAGES_AVIF_QUALITY", 80)
+
+                transform = ImageTransform(size=default_avatar_size)
+                operation = ScaleToPercentOperation(None, default_avatar_size[0])
+                transform = operation.run(transform, avatar)
+                clone_img = willow_image.resize(transform.size)
+
+                img_kwargs = {}
+                if original_format not in ["ico", "svg", "gif", "png"]:
+                    img_kwargs["quality"] = quality
+                if original_format in ["jpeg", "png"]:
+                    img_kwargs["optimize"] = True
+
+                temp_buffer.seek(0)
+
+                # get the method to save the image as based on its format, pass in the args and call it
+                partial(
+                    getattr(clone_img, f"save_as_{original_format}"),
+                    temp_buffer,
+                    **img_kwargs,
+                )()
+
+                # wrap the temp file in a File wrapper
+                reduced_avatar = File(file=temp_buffer, name=avatar.name)
+                return reduced_avatar
```

Comment:
Nit: Unnecessary variable
```suggestion
                return File(file=temp_buffer, name=avatar.name)
```